server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponseTests.java

```diff
@@ -81,34 +85,68 @@ public void testShareResponsesUsingMappingHash() throws Exception {
             inList.add(new FieldCapabilitiesIndexResponse(index, hashing, fieldCaps, canMatch));
         }
         Randomness.shuffle(inList);
-        final List<FieldCapabilitiesIndexResponse> serializedList;
-        try (BytesStreamOutput output = new BytesStreamOutput()) {
-            FieldCapabilitiesIndexResponse.writeList(output, inList);
-            try (
-                StreamInput in = new NamedWriteableAwareStreamInput(
-                    output.bytes().streamInput(),
-                    new NamedWriteableRegistry(Collections.emptyList())
-                )
-            ) {
-                serializedList = FieldCapabilitiesIndexResponse.readList(in);
+        // old versions
+        {
```

Comment:
I had commented earlier, but somehow it was dropped! A tiny comment that this test case is now really long and it might be nice to pull out some set-up logic.

Reply:
++ it would be good to make a separate method for the old versions for instance?

Reply:
Adjusted in https://github.com/elastic/elasticsearch/pull/84596/commits/8657d34012fe4211276183d6a077940070cd8ac2.